### DISCUSSION TITLE
Implement shapes (hidden classes)

### DISCUSSION THREAD

**Discussion by jasonwilliams** [boa-dev/boa#1605] [Ideas]
Great intro suction to the idea: https://mathiasbynens.be/notes/shapes-ics

Shapes provide more optimised storage for objects and arrays and enable fast property access, it might be better to do this after switching over to the VM but worth having a discussion for it now.

**Comment by jedel1043** [comment]
Some more documents that talk about it:

- https://richardartoul.github.io/jekyll/update/2015/04/26/hidden-classes.html
- https://v8.dev/docs/hidden-classes

**Comment by HalidOdat** [comment]
I didn't know there already was a discussion for this that's why I created #2715 :sweat_smile: 

In any case #2723 should resolve this, once fully implemented :)
